include/zephyr/drivers/interrupt_controller/intc_tic.h

```diff
@@ -0,0 +1,205 @@
+/*
+ * Copyright 2024 Hounjoung Rim <hounjoung@tsnlab.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_INCLUDE_DRIVERS_INTERRUPT_CONTROLLER_INTC_TIC_H_
+#define ZEPHYR_INCLUDE_DRIVERS_INTERRUPT_CONTROLLER_INTC_TIC_H_
+
+#include <stdint.h>
+
+#include <zephyr/devicetree.h>
+#include <zephyr/dt-bindings/interrupt-controller/tcc-tic.h>
+#include <zephyr/sys/util_macro.h>
+
+typedef struct tic_distributor {
```

Comment:
This looks awful lot like ARM GICv1, if not identical ... Any reason this cannot use the existing GIC driver?